src/payment/models/buy-crypto/services/buy-crypto.service.ts

```diff
@@ -52,18 +59,40 @@ export class BuyCryptoService {
     return await this.buyCryptoRepo.save(entity);
   }
 
+  async createFromCrypto(cryptoInputId: number, cryptoRouteId: number): Promise<BuyCrypto> {
+    let entity = await this.buyCryptoRepo.findOne({ cryptoInput: { id: cryptoInputId } });
+    if (entity) throw new ConflictException('There is already a buy crypto for the specified crypto input');
+
+    entity = this.buyCryptoRepo.create();
+
+    // crypto input
+    entity.cryptoInput = await this.cryptoInputRepo.findOne(cryptoInputId);
+    if (!entity.cryptoInput) throw new BadRequestException('Crypto input not found');
+
+    // crypto route
+    if (cryptoRouteId) entity.cryptoRoute = await this.getCryptoRoute(cryptoRouteId);
+
+    return await this.buyCryptoRepo.save(entity);
+  }
+
   async update(id: number, dto: UpdateBuyCryptoDto): Promise<BuyCrypto> {
-    let entity = await this.buyCryptoRepo.findOne(id, { relations: ['buy', 'buy.user'] });
+    let entity = await this.buyCryptoRepo.findOne(id, {
+      relations: ['buy', 'buy.user', 'cryptoRoute', 'cryptoRoute.user'],
+    });
     if (!entity) throw new NotFoundException('Buy crypto not found');
 
     const buyIdBefore = entity.buy?.id;
+    const cryptoRouteIdBefore = entity.cryptoRoute?.id;
     const usedRefBefore = entity.usedRef;
 
     const update = this.buyCryptoRepo.create(dto);
 
     // buy
     if (dto.buyId) update.buy = await this.getBuy(dto.buyId);
 
+    // crypto route
+    if (dto.cryptoRouteId) update.cryptoRoute = await this.getCryptoRoute(dto.cryptoRouteId);
```

Comment:
Update of buyId is only allowed, if bankTx is not null, cryptoRouteId otherwise.